trust/trust-check-format.c

```diff
@@ -0,0 +1,458 @@
+        fail("key-value pair is missing a separator ':'");
+        return false;
+    }
+
+    /* check the key */
+    while (line < colon) {
+        if (!isprint(*line)) {
+            fail("key contains non-printable characters");
+            return false;
+        }
+        ++line;
+    }
+    ++line;
+
+    /* skip whitespace after colon */
+    while (isspace(*line))
+        ++line;
+
+    if (*line == '\0') {
+        fail("missing value in key-value pair");
+        return false;
+    }
+
+    /* check if value is a string */
+    value_len = strlen(line);
+    if ((line[0] == '"' && line[value_len - 1] != '"') ||
+        (line[0] != '"' && line[value_len - 1] == '"')) {
+        fail("value string is missing '\"'");
+        return false;
+    }
+
+    /* check the value */
+    for (i = 0; i < value_len; ++i) {
+        if (!isprint(line[i])) {
+            fail("value contains non-printable characters");
+            return false;
+        }
+    }
+
+    return true;
+}
+
+static bool parse_line(char *line) {
+    bool ok;
+    char *end;
+
+    /* check for PEM block */
+    ok = parse_pem(line);
+    if (ok)
+        return true;
+    if (inside_pem)
+        return false;
+
+    /* trim whitespace from both ends */
+    while (isspace(*line))
+        ++line;
+    end = line + strlen(line) - 1;
+    while (line < end && isspace(*end))
+        --end;
+    end[1] = '\0';
+
+    /* ignore empty lines and comments */
+    if (*line == '\0' || *line == '#')
+        return true;
+
+    /* check for section header */
+    if (*line == '[') {
+        ok = parse_section_header(line);
+        if (ok)
+            inside_section = 1;
+        return ok;
+    }
+
+    /* check for key-value pair */
+    ok = parse_key_value_pair(line);
+    if (ok && !inside_section) {
+        fail("key-value pair outside of section");
+        return false;
+    }
+
+    return ok;
+}
+
+/* Checks whether given file has correct .p11-kit format
+ *
+ * Returns 0 when file has correct format,
+ *         1 when file has incorrect format,
+ *         2 when an error has occured
+ */
+static int check_format(const char *filename)
+{
+    bool ok;
+    FILE *stream;
+    char *line;
+
+    stream = fopen(filename, "r");
+    if (stream == NULL) {
+        error("failed to open file for reading");
+        return 2;
+    }
+
+    /* reset context */
+    inside_pem = 0;
+    inside_section = 0;
+
+    for (line_num = 1; (ok = get_line(stream, &line)) && line != NULL; ++line_num) {
+        ok = parse_line(line);
+        free(line);
+        if (!ok) {
+            fclose(stream);
+            return 1;
+        }
+    }
+
+    fclose(stream);
+
+    /* Fail if we received a reading error */
+    if (!ok)
+        return 2;
+
+    /* Fail if PEM block was left open after reaching EOF */
+    if (inside_pem) {
+        fail("PEM block not closed at EOF");
+        return 1;
+    }
+
+    return 0;
+}
+
+int main(int argc, const char **argv)
```

Comment:
Could you update manual in `doc/manual/` (`doc/manual/trust.xml` if we merge this into `trust`)?